plugins/modules/edpm_get_ini_file.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/python3
+# -*- coding: utf-8 -*-
+
+# Copyright 2024 Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import configparser
+
+import yaml
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.0',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = """
+---
+module: edpm_get_ini_file
+author:
+    - Jiri Podivin <jpodivin@redhat.com>
+version_added: '0.0.1'
+short_description: Retrieve contents of section, key or entire given ini file
```

Comment:
as im sure your aware openstack does not use ini files

we use oslo.config format which has some changes form microsfos ini format and is actually a superset

configparser can mostly parse it but you need to be carful with multi_opt strings as that + variable interpolation are the two main extetions to ini format.

Reply:
Actually no, I didn't know that. Even the oslo.config documentation[0] states that it's actually using Ini.
Which I didn't have reason to doubt.

[0] https://docs.openstack.org/oslo.config/yoga/configuration/format.html

Reply:
it using the ini format as a base but its a superset of the format with many extensions that are not part of the standard ini format like variable interpolation, rich typeing fo variables supporting enum, lists, dictionires, multi opts ectra.

basic ini tools can handle the simple case but no OpenStack project that uses oslo.config actually uses ini file.